profiler/profiler.go

```diff
@@ -326,12 +327,18 @@ func (a *agent) createProfile(ctx context.Context) *pb.Profile {
 	var p *pb.Profile
 	md := grpcmd.New(map[string]string{})
 
-	gax.Invoke(ctx, func(ctx context.Context, settings gax.CallSettings) error {
+	err := gax.Invoke(ctx, func(ctx context.Context, settings gax.CallSettings) error {
 		debugLog("creating a new profile via profiler service")
 		var err error
 		p, err = a.client.CreateProfile(ctx, &req, grpc.Trailer(&md))
 		if err != nil {
-			debugLog("failed to create profile, will retry: %v", err)
+			// Certificate errors are permanent errors, agent should
```

Comment:
The comment should mention that gax.Invoke() does not retry the certificate errors (https://github.com/googleapis/gax-go/blob/21fd469b63bc1a2bed9b99b614b4b9dc54a8aeae/v2/invoke.go#L83), that's an important part of why the bug happened.